robot/xarm_env.py

```diff
@@ -1,8 +1,304 @@
+            # In order to fix Kinematics error, if you just force reset to jas, it works
+            # self.robot.set_servo_angle(angle=[3.000007, 15.400017, -91.799985, 76.399969, 4.899992, 0.0, 0.0],
+            #                            wait=True)
+            self.move_xyz(np.array([55.3490479, 2.9007273, 42.4868439]), wait=True)
+            time.sleep(2)
+            self.robot.set_vacuum_gripper(False)
+        else:
+            raise NotImplementedError("Only have one default hardcoded reset pos")
+        if self.random_reset_home_pos:
+            x_magnitude = 0.1
+            y_magnitude = 0.25
+            xyz_delta = np.array([
+                (np.random.random() * 2.0 - 1) * x_magnitude,
+                (np.random.random() * 2.0 - 1) * y_magnitude,
+                0.0
+            ])
+            self.robot.move_xyz(xyz_delta, deltas=True, wait=True)
+            time.sleep(1)
+        self.cur_xyz = self.get_cur_xyz()
+
+    def get_obs(self) -> Dict[str, np.ndarray]:
+        if self.use_camera:
+            self.rgb, self.d = self.cam.get_latest_rgbd()
+        position = self.get_cur_xyz()
+
+        obs = {
+            "ee_pos": position,
+            "delta_ee_pos": (position - self.get_cur_xyz()) / self.scale_factor,
+        }
+
+        if self.use_gripper:
+            obs['ee_pos'] = np.append(obs['ee_pos'], self.gripper_val)
+            obs['delta_ee_pos'] = np.append(obs['delta_ee_pos'], self.gripper_val)
+
+        if self.use_camera:
+            obs['rgb_image'] = self.rgb
+            obs['d_image'] = self.d
+        self.cur_xyz = position
+        return obs
+
+    def _process_obs(self, obs):
+        # For overiding
+        return obs
+
+    def rgb(self) -> np.ndarray:
+        '''
+        return the latest rgb image
+        (useful if you need to query for the last image outside of an RL loop)
+        '''
+        assert self.use_camera
+        return self.rgb
+
+    def move_xyz(self, xyz: np.ndarray, deltas: bool = False, wait: bool = False) -> None:
+        if deltas:
+            xyz = xyz.clip(-1, 1)
+            xyz *= self.scale_factor
+            cur_pos = self.get_cur_xyz()
+            xyz = np.add(cur_pos, xyz)
+        # Clamp it to be within the min and max poses. The min and max are in centimeters
+        # and clamped to centimeters
+        for i in range(len(xyz)):
+            xyz[i] = max(self.min_box[i], xyz[i])
+            xyz[i] = min(self.max_box[i], xyz[i])
+    
+        self.movement_mode()
+
+        message = Float32MultiArray()
+        message.data = np.asarray(xyz*10).copy()
+        self.pub.publish(message)
+
+        self.r.sleep()
+        if wait:
+            self.wait_until_stopped()
+    def get_cur_xyz(self) -> np.ndarray:
+        # In cm
+        error, position = self.robot.get_position()
+        if error != 0:
+            raise NotImplementedError('Need to handle xarm exception')
+        return np.array(position[:3]) / 10
```

Comment:
likewise what's this 10 for? if it's unit conversion add some comment or documentation